src/rviz/default_plugin/odometry_display.h

```diff
@@ -94,14 +94,18 @@ private Q_SLOTS:
   void clear();
 
   void processMessage(const nav_msgs::Odometry::ConstPtr& message) override;
+  bool applyTransforms();
 
   typedef std::deque<rviz::Arrow*> D_Arrow;
   typedef std::deque<rviz::Axes*> D_Axes;
 
   D_Arrow arrows_;
   D_Axes axes_;
 
-  nav_msgs::Odometry::ConstPtr last_used_message_;
+  nav_msgs::Odometry::ConstPtr last_drawn_message_;
+  nav_msgs::Odometry::ConstPtr last_message_;
```

Comment:
What's the difference between those two?

Reply:
last_drawn_message_ is the message which is actually drawn into the scene and required to calculate the positional and angular difference with the current message in order to decide whether the new message should be drawn into the scene or not.
last_message_ was stored to store the timestamp of the most recent message which is required to get the current transform between the fixed frame and the parent frame of the Odometry message.

However, I was able to undo these changes by simply transforming with the most recent transform available (using ros::Time() in transform(), see new commits). The most recent transform should be at least as new as the most recent message due to MessageFilterDisplay. I think this is ok in most scenarios.

Reply:
I could add a warning when a Odometry message with a different parent frame_id arrives, which says that it should be published on another topic.